src/glue/jobs/s3_to_json_s3.py

```diff
@@ -24,97 +25,146 @@
+    with open(data_mapping_fname, "r") as f:
+        data_mapping = json.load(f)
+    logger.debug(f'data_mapping: {data_mapping}')
+    return data_mapping
 
-def parse_client_info_metadata(client_info_str):
-    try:
-        client_info = json.loads(client_info_str)
-        client_info["appVersion"] = str(client_info["appVersion"])
-    except json.JSONDecodeError:
-        app_version_pattern = re.compile(r"appVersion=[^,]+")
-        os_name_pattern = re.compile(r"osName=[^,]+")
-        app_version_search = re.search(app_version_pattern, client_info_str)
-        os_name_search = re.search(os_name_pattern, client_info_str)
-        if app_version_search is None:
-            app_version = None
-            print(client_info_str)
-        else:
-            app_version = app_version_search.group().split("=")[1]
-        if os_name_search is None:
-            os_name = None
-            print(client_info_str)
-        else:
-            os_name = os_name_search.group().split("=")[1]
-        client_info = {
-                "appVersion": app_version,
-                "osName": os_name}
-    return client_info
+def get_archive_map(archive_map_version):
+    archive_map_uri = ARCHIVE_MAP_URI.format(version=archive_map_version)
+    archive_map = requests.get(archive_map_uri)
+    archive_map_json = archive_map.json()
+    return archive_map_json
 
-def process_record(s3_obj, s3_obj_metadata, dataset_mapping):
-    uploaded_on = datetime.strptime(s3_obj_metadata["uploadedon"], '%Y-%m-%dT%H:%M:%S.%fZ')
-    client_info = parse_client_info_metadata(s3_obj_metadata["clientinfo"])
-    logger.info(f"Using dataset mapping for osName = {client_info['osName']} "
-                f"and appVersion = {client_info['appVersion']}")
-    if client_info["osName"] not in dataset_mapping["osName"].keys():
-        logger.warning(f"Skipping {s3_obj_metadata['recordid']} because "
-                       f"osName = {client_info['osName']} was not found "
+def get_json_schema(archive_map, assessment_id, assessment_revision, file_name):
+    # First check universally used files
+    for file in archive_map["anyOf"]:
+        if file["filename"] == file_name:
+            if "deprecated" in file \
+                    and file["deprecated"] is True \
+                    and "jsonSchema" not in file:
+                continue
+            json_schema = requests.get(file["jsonSchema"])
+            return json_schema.json()
+    # Next check app-specific files
+    for app in archive_map["apps"]:
+        if app["appId"] == "mobile-toolbox":
+            is_valid_assessment = any([
+                    a["assessmentIdentifier"] == assessment_id
+                    and a["assessmentRevision"] == assessment_revision
+                    for a in app["assessments"]])
+            if is_valid_assessment:
+                for default_file in app["default"]["files"]:
+                    if default_file["filename"] == file_name:
```

Comment:
No need to do same deprecated check here?

Reply:
I looked deeper into the schema for archive-map.json and it turns out that all of these file arrays use the same FileInfo object schema. So I decided to refactor this function and that turned into a refactor of the entire script.

To answer your question -- no. And when I refactored this I wrote a function for getting a JSON Schema from any FileInfo object and left out any deprecation check. We could be processing any data at any time, whether that data has been deprecated or not, so it shouldn't make a difference to us if something is deprecated. According to Shannon the "deprecated" flag was only meant to be used to mark old mPower 2 files like info.json and taskResult.json, and those don't have JSON Schema included in archive-map.json... long story short, the "deprecated" flag is irrelevant to us.